[Tests] TestRemoteQueryCommunity.test_remote_select_torrents FAILURE

https://github.com/Tribler/tribler/actions/runs/3367858797/jobs/5585761167 in https://github.com/Tribler/tribler/pull/7112

```python
src/tribler/core/components/metadata_store/remote_query_community/tests/test_remote_query_community.py . [ 15%]
Exception ignored in: <Finalize object, dead>
................F
Traceback (most recent call last):

  File "/Users/runner/hostedtoolcache/Python/3.8.14/x[64](https://github.com/Tribler/tribler/actions/runs/3367858797/jobs/5585761167#step:5:65)/lib/python3.8/multiprocessing/util.py", line 224, in __call__
=================================== FAILURES ===================================
    res = self._callback(*self._args, **self._kwargs)
_____________ TestRemoteQueryCommunity.test_remote_select_torrents _____________
  File "/Users/runner/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/multiprocessing/util.py", line 464, in close_fds

    os.close(fd)
self = <tribler.core.components.metadata_store.remote_query_community.tests.test_remote_query_community.TestRemoteQueryCommunity testMethod=test_remote_select_torrents>
OSError: [Errno 9] Bad file descriptor

    async def test_remote_select_torrents(self):
        """
        Test dropping packets that go over the response limit for a remote select.
    
        """
        peer = self.nodes[0].my_peer
        mds0 = self.nodes[0].overlay.mds
        mds1 = self.nodes[1].overlay.mds
    
        with db_session:
            chan = mds0.ChannelMetadata.create_channel(random_string(), "")
            torrent_infohash = random_infohash()
            torrent = mds0.TorrentMetadata(origin_id=chan.id_, infohash=torrent_infohash, title='title1')
            torrent.sign()
    
        processing_results = []
    
        def callback(request, results):  # pylint: disable=unused-argument
            processing_results.extend(results)
    
        self.nodes[1].overlay.send_remote_select(
            peer, metadata_type=[REGULAR_TORRENT], infohash=torrent_infohash, processing_callback=callback
        )
        await self.deliver_messages()
    
>       assert len(processing_results) == 1
E       assert 0 == 1
E        +  where 0 = len([])

src/tribler/core/components/metadata_store/remote_query_community/tests/test_remote_query_community.py:254: AssertionError
------------------------------ Captured log call -------------------------------
08:16:50.100 INFO BasicRemoteQueryCommunity(13) Init. Settings: None.
08:16:50.191 INFO BasicRemoteQueryCommunity(13) Init. Settings: None.
08:16:54.527 INFO EVA(403) Shutting down...
08:16:54.529 INFO EVA(416) Shutting down completed
08:16:54.5[68](https://github.com/Tribler/tribler/actions/runs/3367858797/jobs/5585761167#step:5:69) INFO EVA(403) Shutting down...
08:16:54.5[72](https://github.com/Tribler/tribler/actions/runs/3367858797/jobs/5585761167#step:5:73) INFO EVA(416) Shutting down completed
=========================== short test summary info ============================
```

--- a/src/tribler/core/components/metadata_store/remote_query_community/tests/test_remote_query_community.py
+++ b/src/tribler/core/components/metadata_store/remote_query_community/tests/test_remote_query_community.py
@@ -1,3 +1,4 @@
+import asyncio
 import random
 import string
 import time
@@ -7,6 +8,7 @@ from operator import attrgetter
 from os import urandom
 from unittest.mock import Mock, patch
 
+import pytest
 from ipv8.keyvault.crypto import default_eccrypto
 from ipv8.test.base import TestBase
 from pony.orm import db_session
@@ -225,6 +227,7 @@ class TestRemoteQueryCommunity(TestBase):
         Test if sending back information on updated version of a metadata entry works
         """
 
+    @pytest.mark.timeout(10)
     async def test_remote_select_torrents(self):
         """
         Test dropping packets that go over the response limit for a remote select.
@@ -240,15 +243,18 @@ class TestRemoteQueryCommunity(TestBase):
             torrent = mds0.TorrentMetadata(origin_id=chan.id_, infohash=torrent_infohash, title='title1')
             torrent.sign()
 
+        callback_called = asyncio.Event()
         processing_results = []
 
-        def callback(request, results):  # pylint: disable=unused-argument
+        def callback(_, results):
             processing_results.extend(results)
+            callback_called.set()
 
         self.nodes[1].overlay.send_remote_select(
             peer, metadata_type=[REGULAR_TORRENT], infohash=torrent_infohash, processing_callback=callback
         )
-        await self.deliver_messages()
+
+        await callback_called.wait()
 
         assert len(processing_results) == 1
         obj = processing_results[0].md_obj
@@ -263,10 +269,13 @@ class TestRemoteQueryCommunity(TestBase):
             torrent.sign()
 
         processing_results = []
+        callback_called.clear()
+
         self.nodes[1].overlay.send_remote_select(
             peer, metadata_type=[REGULAR_TORRENT], infohash=torrent_infohash, processing_callback=callback
         )
-        await self.deliver_messages()
+
+        await callback_called.wait()
 
         assert len(processing_results) == 1
         obj = processing_results[0].md_obj
@@ -597,6 +606,7 @@ class TestRemoteQueryCommunity(TestBase):
         b.send_remote_select(peer_a, **kwargs2, processing_callback=callback2)
 
         original_get_entries = MetadataStore.get_entries
+
         # Add a delay to ensure that the first query is still being processed when the second one arrives
         # (the mds.get_entries() method is a synchronous one and is called from a worker thread)
 

